### ISSUE TITLE
snap: Barrier can't remember settings

### ISSUE THREAD

**Issue by ghost** [input-leap/input-leap#454]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/454
> 
> Issue created on: 2019-10-08 by @gapplef
> Issue last updated on: 2021-01-09

### Operating Systems ###
Server: Ubuntu Desktop 19.10
Client: Ubuntu Desktop 18.04

### Barrier Version ###
2.3.2-snapshot-210c2b70

I checked `Minimize to System Tray` & `Hide on startup` in Settings, and set Barrier to autostart with Gnome Tweaks.

1. Barrier show up after system restart and the `Hide on startup` options become unchecked.
I checked by exchange the server/client and found this problem only appear on Ubuntu 19.10, no matter it's server or client.
This problem can be solved by disable autostart.

2. After change the server IP or change from client to server, or vice versa, Barrier must be quit manually before the first time system restart, otherwise the change will be reset on restart.
This problem appear on both Ubuntu 18.04 and Ubuntu 19.10.
This problem not related with autostart.

---

> Commented on: 2020-06-11 by @mim0

I had the same problem, but I solved it by saving the settings to a file.

You can do it through the application toolbar menu: Barrier -> Save configuration
Or using the Shortcut: Ctrl + Alt + S

After that, it always remembers my config.

---

> Commented on: 2021-01-09 by @pc-2198x

**Operating System**
Arch Linux 5.10.5-arch1-1 

**Barrier Version**
2.3.3

I also checked `Minimize to System Tray` as well as `Hide on startup` and generated a configuration file by using `Ctrl + Alt + S`.
However, when Barrier boots, it pulls config from `/temp`, not my `/Home` directory as expected:

`[2021-01-09T12:02:03] INFO: config file: /tmp/Barrier.ALChYC`

**Comment by shymega** [comment]
I don't think this is Input Leap's fault - I think it's likely that Snap is restricting access to the directory in `$HOME`. Sandboxing and all. Also, we don't provide a Snap yet, so this issue is yet to be revisited.

**Comment by mzryz** [comment]
I'm not using Snap, I'm using Barrier from the Manjaro repo and it isn't remembering settings either - although it *does* on the Plasma desktop, just not Gnome... I see this has been an issue since 2019 - how can such a basic functionality issue not be solved in four years... I wonder.

**Comment by shymega** [comment]
@mzryz Thank you for reporting it's current on Manjaro natively. Hm, you wonder? Well, we don't get paid for this project, it's strictly voluntary - I apologise profusely that we don't offer premium support and development... but that's open source. Feel free to make a PR to fix it. 🙄
